test/media/segment_index_unit.js

```diff
@@ -955,9 +955,11 @@ describe('SegmentIndex', /** @suppress {accessControls} */ () => {
    * @param {number} startTime
    * @param {number} endTime
    * @param {!Array.<!shaka.media.SegmentReference>=} partialReferences
+   * @param {!shaka.extern.HlsAes128Key} hlsAes128Key
    * @return {shaka.media.SegmentReference}
    */
-  function makeReference(uri, startTime, endTime, partialReferences = []) {
+  function makeReference(uri, startTime, endTime, partialReferences = [],
+      hlsAes128Key = {method: 'AES-128', firstMediaSequenceNumber: 0}) {
```

Comment:
Do we want to default that to non-null at all times in all tests?

Could you instead make a regression test for this particular bug, where you show that hlsAes128Key is preserved on the last segment after a fit() call?

Reply:
Done!

Reply:
## Before

![2022-10-03 01-10-03](https://user-images.githubusercontent.com/23169202/193465732-f8c2cf1b-fb78-45f2-8e9a-0bfc73d67635.png)

## After

![2022-10-03 01-10-45](https://user-images.githubusercontent.com/23169202/193465738-a816bbdc-2d30-4f84-85ba-d507bdb11863.png)